### PR TITLE
Fixes #959:Suggestion box closes on clicking outside of it.

### LINKED ISSUE #959: Suggestion box do not closes on clicking outside it.
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behavior:**
Suggestion box do not closes on clicking outside it.

**Expected behavior:**
It should be closed on clicking outside it.

**Screenshots of the issue:**
bug

I will fix it soon.

### PR DESCRIPTION
**Pull request by praveenojha33**
Fixes #959 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

https://pr-961-fossasia-susper.surge.sh

#### Changes proposed in this pull request:
I have fixed #959  Suggestion box now closes automatically when clicking outside of it.

### CODE REVIEW

File: src/app/search-bar/search-bar.component.ts [hunk 1]
```diff
@@ -14,7 +14,10 @@ import * as speechactions from '../actions/speech';
 @Component({
   selector: 'app-search-bar',
   templateUrl: './search-bar.component.html',
-  styleUrls: ['./search-bar.component.css']
+  styleUrls: ['./search-bar.component.css'],
+  host: {
+'(document:click)': 'sugClose($event)'
```
Comment (harshit98) [inline src/app/search-bar/search-bar.component.ts]: @praveenojha33 Kindly indent the code here properly.

File: src/app/search-bar/search-bar.component.ts [hunk 2]
```diff
@@ -74,13 +78,17 @@ export class SearchBarComponent implements OnInit, AfterViewInit {
...
-      this.displayStatus = 'hidebox';
+      this.displayStatus = 'showbox';
       event.target.blur();
       event.preventDefault();
       this.submit();
     }
   }
+  sugClose(event) {
+    if (!this._eref.nativeElement.contains(event.target)) {
+  this.displayStatus = 'hidebox';
```
Comment (harshit98) [inline src/app/search-bar/search-bar.component.ts]: Indent the code.

File: src/app/search-bar/search-bar.component.ts [hunk 3]
```diff
@@ -74,13 +78,17 @@ export class SearchBarComponent implements OnInit, AfterViewInit {
...
+}
+  }
```
Comment (harshit98) [inline src/app/search-bar/search-bar.component.ts]: Same goes here.

File: tslint.json [hunk 4]
```diff
@@ -92,7 +92,7 @@
     "component-selector": [true, "element", "app", "kebab-case"],
     "use-input-property-decorator": true,
     "use-output-property-decorator": true,
-    "use-host-property-decorator": true,
+    "use-host-property-decorator": false,
```
Comment (harshit98) [inline tslint.json]: What's the purpose of it?

Comment (praveenojha33) [reply tslint.json]: Actually I am using `host` property to listen for click event on document and then I am directing it to `sugClose($event)` function to tell what is to be done. But then it results in linting error. It asks me to use Host Property Decorators i.e `@HostBindings` and `@HostListeners` to do the same task. Then I followed this answer https://stackoverflow.com/questions/45172565/use-hostbindings-instead-host-in-angular-4 and I came to know that it is fine to use host property. So I turned it to false. But now it is decreasing the code coverage.

File: src/app/search-bar/search-bar.component.ts [hunk 5]
```diff
@@ -14,7 +14,10 @@ import * as speechactions from '../actions/speech';
 @Component({
   selector: 'app-search-bar',
   templateUrl: './search-bar.component.html',
-  styleUrls: ['./search-bar.component.css']
+  styleUrls: ['./search-bar.component.css'],
+  host: {
+          '(document:click)': 'sugClose($event)'
+        }
```
Comment (harshit98) [inline src/app/search-bar/search-bar.component.ts]: Please follow the coding style properly.

Comment (praveenojha33) [reply src/app/search-bar/search-bar.component.ts]: I have indented the code, what should I do now ?

**Pull request closed without merging**